### PR TITLE
add dual-stack support for Azure

### PR DESCRIPTION
**Pull request by PratikDeoghare**
**What this PR does / why we need it**:
 Adds dual-stack support for Azure.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>` format, will close the issue(s) when PR gets merged)*:
Implements part of https://github.com/kubermatic/kubermatic/issues/9351

**Special notes for your reviewer**:

**Optional Release Note**:

```release-note
 add dual-stack support for Azure.
```

### CODE REVIEW

File: pkg/cloudprovider/provider/azure/provider.go [hunk 1]
```diff
@@ -434,6 +435,16 @@ func getNICIPAddresses(ctx context.Context, c *config, ifaceName string) (map[st
 				for _, ip := range publicIPs {
 					ipAddresses[ip] = v1.NodeExternalIP
 				}
+
+				publicIP6Name := ifaceName + "-pubipv6"
```
Comment (rastislavs) [inline pkg/cloudprovider/provider/azure/provider.go:439]: nit: may be a good idea to use constants for these interface name suffixes (`-pubip`, `-pubipv6`)

File: pkg/cloudprovider/provider/azure/provider.go [hunk 2]
```diff
@@ -561,9 +572,12 @@ func (p *provider) Create(machine *clusterv1alpha1.Machine, data *cloudprovidert
 		return nil, fmt.Errorf("failed to generate ssh key: %v", err)
 	}
 
-	ifaceName := machine.Name + "-netiface"
-	publicIPName := ifaceName + "-pubip"
-	var publicIP *network.PublicIPAddress
+	netFamily := providerCfg.Network.GetNetworkFamily()
+	sku := network.PublicIPAddressSkuNameBasic
+	if netFamily == util.DualStack {
+		sku = network.PublicIPAddressSkuNameStandard
```
Comment (rastislavs) [inline pkg/cloudprovider/provider/azure/provider.go:578]: maybe leave a comment here on why we need the Standard SKU

**Review by moadqassem — APPROVED** [review]
/approve
/lgtm

**Pull request merged**